### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by bryantChhun**
This PR adds detailed documentation for the pipeline as it was used for the `dynamorph` paper:

[https://www.biorxiv.org/content/10.1101/2020.07.20.213074v1](https://www.biorxiv.org/content/10.1101/2020.07.20.213074v1)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,18 +19,23 @@ This is a repo storing codes and scripts for **DynaMorph: learning morphodynamic
...
+- [imageio](https://imageio.github.io/) 
+- [tifffile](https://pypi.org/project/tifffile/) 
+- [Matplotlib](https://matplotlib.org/) 
+- [OpenCV](https://opencv.org/about/) 
+- [PyTorch](https://pytorch.org/) 
+- [SciPy](https://www.scipy.org/) 
+- [scikit-learn](https://scikit-learn.org/) 
+- [umap-learn](https://umap-learn.readthedocs.io/en/latest/#)
+- [pyyaml](https://pyyaml.org/)
+- [h5py](https://docs.h5py.org/en/stable/)
```
Comment (smguo) [inline README.md:36]: Minor point: It looks like the code base doesn't use h5py and POT anymore. We should remove the dependency in the next refactor

File: README.md [hunk 2]
```diff
@@ -19,18 +19,23 @@ This is a repo storing codes and scripts for **DynaMorph: learning morphodynamic
...
+- [POT](https://pythonot.github.io/)
+
 
 ## Getting Started
```
Comment (smguo) [inline README.md:56]: I think this section should be installation guide

File: documents/1-preprocessing.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+- dimensions of (T, Y, X)
+
+
+In both cases, there is a check that every image of the series has the same X-Y dimensions
+
+## outupt
+ 
+The output file will be:
+
+- a single .npy of shape `(T, C, Y, X)`
```
Comment (smguo) [inline documents/1-preprocessing.md]: Should be 5D array. Will submit a PR to edit this.

File: documents/5-dim_reduction.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
+# dimensionality reduciton
```
Comment (smguo) [inline documents/5-dim_reduction.md]: Some of the markdown doesn't seem to render properly on Github. Not sure why...

**Review by smguo — APPROVED** [review]
Looks great! I left some minor comments and submitted a PR to this branch to add some edits.

File: README.md (hunk 1 above)
Comment (bryantChhun) [reply README.md:36]: Sounds good

File: README.md (hunk 2 above)
Comment (bryantChhun) [reply README.md:56]: I added a new section above that is #installation

**Pull request merged**